src/connectors/source.rs

```diff
@@ -834,21 +840,24 @@ where
     async fn handle_link(
         &mut self,
         port: Cow<'static, str>,
+        tx: Sender<Result<()>>,
         mut pipelines: Vec<(DeployEndpoint, pipeline::Addr)>,
-    ) -> Control {
+    ) -> Result<Control> {
         let pipes = if port.eq_ignore_ascii_case(OUT.as_ref()) {
             &mut self.pipelines_out
         } else if port.eq_ignore_ascii_case(ERR.as_ref()) {
             &mut self.pipelines_err
         } else {
             error!("{} Tried to connect to invalid port: {}", &self.ctx, &port);
-            return Control::Continue;
+            return Ok(Control::Continue);
```

Comment:
make sure that in every case, be it error or success, we send a `Result<()>` to the `tx`. One idea to simplify this would be to have this function return a `Result<()>` that can then be sent via the `tx` and it doesn't have to be passed in here. Just call the function above, await the result and send it away. This way one response is guaranteed.